payjoin/src/receive/mod.rs

```diff
@@ -740,21 +745,27 @@ impl ProvisionalProposal {
 
     /// Calculate the additional input weight contributed by the receiver
     fn additional_input_weight(&self) -> Result<Weight, RequestError> {
-        // This error should never happen. We check for at least one input in the constructor
-        let input_pair = self
-            .payjoin_psbt
-            .input_pairs()
-            .next()
-            .ok_or(InternalRequestError::OriginalPsbtNotBroadcastable)?;
-        // Calculate the additional weight contribution
-        let input_count = self.payjoin_psbt.inputs.len() - self.original_psbt.inputs.len();
-        log::trace!("input_count : {}", input_count);
-        let weight_per_input =
-            input_pair.expected_input_weight().map_err(InternalRequestError::InputWeight)?;
-        log::trace!("weight_per_input : {}", weight_per_input);
-        let contribution_weight = weight_per_input * input_count as u64;
-        log::trace!("contribution_weight: {}", contribution_weight);
-        Ok(contribution_weight)
+        let payjoin_inputs_weight = self.payjoin_psbt.input_pairs().try_fold(
+            Weight::ZERO,
+            |acc, input_pair| -> Result<Weight, RequestError> {
+                let input_weight = input_pair
+                    .expected_input_weight()
+                    .map_err(InternalRequestError::InputWeight)?;
+                Ok(acc + input_weight)
+            },
+        )?;
+        let original_inputs_weight = self.original_psbt.input_pairs().try_fold(
+            Weight::ZERO,
+            |acc, input_pair| -> Result<Weight, RequestError> {
+                let input_weight = input_pair
+                    .expected_input_weight()
+                    .map_err(InternalRequestError::InputWeight)?;
+                Ok(acc + input_weight)
+            },
+        )?;
```

Comment:
Duplication makes me think a helper would make this even prettier. I might even define the function inside additional_input_weight since it's only used there (is that style frowned upon?)

```suggestion
        fn input_weight(psbt: &Psbt) -> Result<Weight, RequestError> {
            psbt.input_pairs().try_fold(
                Weight::ZERO,
                |acc, input_pair| -> Result<Weight, RequestError> {
                    let input_weight = input_pair
                        .expected_input_weight()
                        .map_err(InternalRequestError::InputWeight)?;
                    Ok(acc + input_weight)
                },
            )
        }

        let payjoin_inputs_weight = input_weight(&self.payjoin_psbt)?;
        let original_inputs_weight = input_weight(&self.original_psbt)?;
```